GVRf/Framework/framework/src/main/jni/engine/renderer/batch.cpp

```diff
@@ -0,0 +1,197 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "batch.h"
+#include "objects/scene_object.h"
+#include "objects/components/camera.h"
+#include "objects/components/render_data.h"
+#include "objects/light.h"
+
+#define BATCH_SIZE 60
+
+namespace gvr {
+
+Batch::Batch() :
+        draw_count_(0), vertex_count_(0), index_count_(0), vertex_limit_(400),
+        indices_limit_(400), renderdata_(nullptr), index_offset_(0), not_batched_(false) {
+    vertices_.clear();
+    normals_.clear();
+    tex_coords_.clear();
+    indices_.clear();
+    matrices_.clear();
+    matrix_indices_.clear();
```

Comment:
If common inititalisation in done in the 2 ctors, better to move them to a function called in both ctors.

Also, it this necessary?